Dart/src/com/jetbrains/lang/dart/ide/formatter/DartSpacingProcessor.java

```diff
@@ -404,9 +409,47 @@ private Spacing setBraceSpace(boolean needSpaceSetting,
       return Spacing.createDependentLFSpacing(spaces, spaces, textRange, mySettings.KEEP_LINE_BREAKS, mySettings.KEEP_BLANK_LINES_IN_CODE);
     }
     else {
-      final int lineBreaks = braceStyleSetting == CommonCodeStyleSettings.END_OF_LINE ||
-                             braceStyleSetting == CommonCodeStyleSettings.NEXT_LINE_IF_WRAPPED ? 0 : 1;
+      final int lineBreaks =
+        braceStyleSetting == CommonCodeStyleSettings.END_OF_LINE || braceStyleSetting == CommonCodeStyleSettings.NEXT_LINE_IF_WRAPPED
+        ? 0
+        : 1;
       return Spacing.createSpacing(spaces, spaces, lineBreaks, false, 0);
     }
   }
+
+  private boolean allCascadesAreSameMethod(ASTNode[] children) {
+    for (int i = 1; i < children.length; i++) {
+      if (!cascadesAreSameMethod(children[i - 1], children[i])) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  private boolean cascadesAreSameMethod(ASTNode child1, ASTNode child2) {
+    ASTNode call1 = child1.getLastChildNode();
+    if (call1.getElementType() == CALL_EXPRESSION) {
+      ASTNode call2 = child2.getLastChildNode();
+      if (call2.getElementType() == CALL_EXPRESSION) {
+        String name1 = getImmediateCallName(call1);
+        if (name1 != null) {
+          String name2 = getImmediateCallName(call2);
+          if (name1.equals(name2)) {
+            return true;
+          }
+        }
+      }
+    }
+    return false;
+  }
+
+  private String getImmediateCallName(ASTNode callNode) {
```

Comment:
I track warnings in the following way:
- Colored square in the editor top right shows file status. When it's yellow I try to make sure that warnings are justified. It is possible to suppress justified warning to make inspection skip it and show green file status
- Yellow stripes on the right gutter
- The fastest way to navigate between warnings is F2 / Shift+F2

Speaking about this warning I browsed internet a bit and noticed that most people recommend to make helper methods static. Special highlighting in IDE allows to distinguish static / dynamic methods calls. Seeing a static invocation I'm sure that this method doesn't need to read and won't change this instance state. One of more detailed answers [here](http://stackoverflow.com/questions/3346764/should-java-methods-be-static-by-default).